From patchwork Fri Nov 22 10:58:22 2024
Subject: [f2fs-dev] [PATCH V2] f2fs: cache more dentry pages

While traversing dir entries in dentry page, it's better to refresh current
accessed page in lru list by using FGP_ACCESSED flag, otherwise, such page
may has less chance to survive during memory reclaim, result in causing
additional IO when revisiting dentry page.

diff --git a/fs/f2fs/data.c b/fs/f2fs/data.c
index 402e00d54c0b..71ca987fe7cd 100644
--- a/fs/f2fs/data.c
+++ b/fs/f2fs/data.c
@@ -1286,7 +1286,7 @@ struct page *f2fs_find_data_page(struct inode *inode, pgoff_t index,
        struct address_space *mapping = inode->i_mapping;
        struct page *page;

-       page = find_get_page(mapping, index);
+       page = find_get_page_flags(mapping, index, FGP_ACCESSED);
        if (page && PageUptodate(page))
                return page;
        f2fs_put_page(page, 0);
